packages/snaps-utils/src/ui.tsx

```diff
@@ -0,0 +1,411 @@
+ *
+ * @param variant - The legacy button component variant.
+ * @returns The button variant.
+ */
+function getButtonVariant(variant?: 'primary' | 'secondary' | undefined) {
+  switch (variant) {
+    case 'primary':
+      return 'primary';
+    case 'secondary':
+      return 'destructive';
+    default:
+      return undefined;
+  }
+}
+
+/**
+ * Get the children of a JSX element. If there is only one child, the child is
+ * returned directly. Otherwise, the children are returned as an array.
+ *
+ * @param elements - The JSX elements.
+ * @returns The child or children.
+ */
+function getChildren<Type>(elements: Type[]) {
+  if (elements.length === 1) {
+    return elements[0];
+  }
+
+  return elements;
+}
+
+/**
+ * Get the text of a link token.
+ *
+ * @param token - The link token.
+ * @returns The text of the link token.
+ */
+function getLinkText(token: Tokens.Link | Tokens.Generic) {
+  if (token.tokens && token.tokens.length > 0) {
+    return getChildren(token.tokens.flatMap(getTextChildFromToken));
+  }
+
+  return token.href;
+}
+
+/**
+ * Get the text child from a list of markdown tokens.
+ *
+ * @param tokens - The markdown tokens.
+ * @returns The text child.
+ */
+function getTextChildFromTokens(tokens: Token[]) {
+  return getChildren(tokens.flatMap(getTextChildFromToken));
+}
+
+/**
+ * Get the text child from a markdown token.
+ *
+ * @param token - The markdown token.
+ * @returns The text child.
+ */
+function getTextChildFromToken(token: Token): TextChildren {
+  switch (token.type) {
+    case 'link': {
+      return <Link href={token.href} children={getLinkText(token)} />;
+    }
+
+    case 'text':
+      return token.text;
+
+    case 'strong':
+      return (
+        <Bold>
+          {
+            getTextChildFromTokens(
+              // Due to the way `marked` is typed, `token.tokens` can be
+              // `undefined`, but it's a required field of `Tokens.Bold`, so we
+              // can safely cast it to `Token[]`.
+              token.tokens as Token[],
+            ) as BoldChildren
+          }
+        </Bold>
+      );
+
+    case 'em':
+      return (
+        <Italic>
+          {
+            getTextChildFromTokens(
+              // Due to the way `marked` is typed, `token.tokens` can be
+              // `undefined`, but it's a required field of `Tokens.Bold`, so we
+              // can safely cast it to `Token[]`.
+              token.tokens as Token[],
+            ) as ItalicChildren
+          }
+        </Italic>
+      );
+
+    default:
+      return null;
+  }
+}
+
+/**
+ * Get all text children from a markdown string.
+ *
+ * @param value - The markdown string.
+ * @returns The text children.
+ */
+export function getTextChildren(value: string) {
+  const rootTokens = lexer(value, { gfm: false });
+  const children: TextChildren = [];
+
+  walkTokens(rootTokens, (token) => {
+    if (token.type === 'paragraph') {
+      if (children.length > 0) {
+        children.push('\n\n');
```

Comment:
?

Reply:
This inserts an empty line between each paragraph.

Reply:
Doesn't this add an empty line at the top as well?

Reply:
No, that's what the `children.length > 0` is for.